src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,123 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.model.ValidationException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final String VALID_LOGIN = "Valid login";
+    private static final String VALID_PASSWORD = "Valid password";
+    private static final String INVALID_BORDERLINE_LOGIN = "user1";
+    private static final String INVALID_BORDERLINE_PASSWORD = "12345";
+    private static final int VALID_AGE = 18;
+    private static final int INVALID_BORDERLINE_AGE = 17;
+    private static final int INVALID_NEGATIVE_AGE = -2;
+    private RegistrationServiceImpl registrationService;
+    private StorageDaoImpl storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_checkUserExist_notOk() {
+        User registeredUser = new User();
+        registeredUser.setLogin(VALID_LOGIN);
+        registeredUser.setPassword(VALID_PASSWORD);
+        registeredUser.setAge(VALID_AGE);
+        storageDao.add(registeredUser);
+        User newUser = new User();
+        newUser.setLogin(VALID_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setAge(VALID_AGE);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("User with your login has already exists.Try new login!",
+                exception.getMessage());
```

Comment:
It seems like the test case `register_checkUserExist_notOk` is supposed to check if a duplicate user cannot be registered, but there's no implementation code provided to show that this check is actually happening in the `RegistrationServiceImpl`. Make sure such a validation is implemented and tested.